extra_data/components.py

```diff
@@ -1461,9 +1464,19 @@ class JUNGFRAU(MultimodDetectorBase):
     module_shape = (512, 1024)
 
     def __init__(self, data: DataCollection, detector_name=None, modules=None,
-                 *, min_modules=1, n_modules=None):
+                 *, min_modules=1, n_modules=None, st_modno=1):
         super().__init__(data, detector_name, modules, min_modules=min_modules)
 
+        if st_modno:
+            self.modno_to_source = {}
+            # Overwrite modno based on given starting module number and update
+            # source_to_modno and modno_to_source.
+            for source in self.source_to_modno.keys():
+                # JUNGFRAU modono is expected (e.g. extra_geom) to start with 1.
+                modno = int(self._source_re.search(source)['modno']) - st_modno + 1
+                self.source_to_modno[source] = modno
+                self.modno_to_source[modno] = source
+
         if n_modules is not None:
             self.n_modules = int(n_modules)
         else:
```

Comment:
I think this branch (estimating `n_modules`) needs to be adjusted too.

Reply:
Nice catch. I fixed this branch when `n_modules is None` [77b88ec9ed1944e746bcd979a917e0f10d45abf6](https://github.com/European-XFEL/EXtra-data/pull/379/commits/77b88ec9ed1944e746bcd979a917e0f10d45abf6)